TypeScripter.Common/Generators/DataServiceGenerator.cs

```diff
@@ -135,39 +140,49 @@ public static List<string> Generate(string apiRelativePath, List<Type> apiContro
 			}
 
 			sb.AppendLine();
-			sb.AppendLine(HttpModule == Http ? "\tprivate handleError(error: Response | any) {" : "\tprivate handleError(error: HttpErrorResponse | any) {");
-			sb.AppendLine("\t\tlet errMsg: string;");
-			if (HttpModule == Http)
-			{
-				sb.AppendLine("\t\tif (error instanceof Response) {");
-				sb.AppendLine("\t\t\tconst hdr = error.headers.get('ExceptionMessage');");
-				sb.AppendLine("\t\t\tconst body = error.json() || '';");
-				sb.AppendLine("\t\t\tconst err = body.error || JSON.stringify(body);");
-				sb.AppendLine("\t\t\terrMsg = `${error.status}${hdr ? ' - ' + hdr : ''} - ${error.statusText || ''} ${err}`;");
-				sb.AppendLine("\t\t} else {");
-				sb.AppendLine("\t\t\terrMsg = error.message ? error.message : error.toString();");
-				sb.AppendLine("\t\t}");
-			}
-			else
-			{
-				sb.AppendLine("\t\tif (error instanceof HttpErrorResponse) {");
-				sb.AppendLine("\t\t\tconst hdr = error.headers.get('ExceptionMessage');");
-				sb.AppendLine("\t\t\terrMsg = `${error.status}${hdr ? ' - ' + hdr : ''} - ${error.statusText || ''} ${error.error}`;");
-				sb.AppendLine("\t\t} else {");
-				sb.AppendLine("\t\t\terrMsg = error.message ? error.message : error.toString();");
-				sb.AppendLine("\t\t}");
-			}
-			sb.AppendLine("\t\tconsole.error(errMsg);");
-			sb.AppendLine("\t\treturn Observable.throw(errMsg);");
-			sb.AppendLine("\t}");
+			sb.AppendLine(httpModule == Http ? "\tprivate handleError(error: Response | any) {" : "\tprivate handleError(error: HttpErrorResponse | any) {");
+		  if (handleErrors)
+		  {
+		    sb.AppendLine("\t\tlet errMsg: string;");
+		    if (httpModule == Http)
+		    {
+		      sb.AppendLine("\t\tif (error instanceof Response) {");
+		      sb.AppendLine("\t\t\tconst hdr = error.headers.get('ExceptionMessage');");
+		      sb.AppendLine("\t\t\tconst body = error.json() || '';");
+		      sb.AppendLine("\t\t\tconst err = body.error || JSON.stringify(body);");
+		      sb.AppendLine("\t\t\terrMsg = `${error.status}${hdr ? ' - ' + hdr : ''} - ${error.statusText || ''} ${err}`;");
+		      sb.AppendLine("\t\t} else {");
+		      sb.AppendLine("\t\t\terrMsg = error.message ? error.message : error.toString();");
+		      sb.AppendLine("\t\t}");
+		    }
+		    else
+		    {
+		      sb.AppendLine("\t\tif (error instanceof HttpErrorResponse) {");
+		      sb.AppendLine("\t\t\tconst hdr = error.headers.get('ExceptionMessage');");
+		      sb.AppendLine(
+		        "\t\t\terrMsg = `${error.status}${hdr ? ' - ' + hdr : ''} - ${error.statusText || ''} ${error.error}`;");
+		      sb.AppendLine("\t\t} else {");
+		      sb.AppendLine("\t\t\terrMsg = error.message ? error.message : error.toString();");
+		      sb.AppendLine("\t\t}");
+		    }
+
+		    sb.AppendLine("\t\tconsole.error(errMsg);");
+		    sb.AppendLine("\t\treturn throwError(errMsg);");
```

Comment:
@benhaynie we spoke about this, I figured I was here and would fix it. Hopefully doesn't conflict with anything you were doing.